core/src/main/java/org/apache/iceberg/avro/AvroFileAppender.java

```diff
@@ -85,7 +86,7 @@ public long length() {
         throw new RuntimeIOException(e, "Failed to get stream length");
       }
     }
-    throw new RuntimeIOException("Failed to get stream length: no open stream");
+    throw new FileHandlingException("Failed to get stream length: no open stream");
```

Comment:
Could we change this to a precondition check as opposed to file exception?

Reply:
ACK. Changed it.